Example/ViewController+FreeDrive.swift

```diff
@@ -76,24 +76,37 @@ extension ViewController {
     }
     
     func updateFreeDriveStyledFeatures() {
-        _ = navigationMapView.mapView.style.updateGeoJSON(for: trackStyledFeature.sourceIdentifier,
-                                                          with: Feature(geometry: .lineString(trackStyledFeature.lineString)))
-        
-        _ = navigationMapView.mapView.style.updateGeoJSON(for: rawTrackStyledFeature.sourceIdentifier,
-                                                          with: Feature(geometry: .lineString(rawTrackStyledFeature.lineString)))
+        do {
+            let style = navigationMapView.mapView.style
+            if let _ = try style?.source(withId: trackStyledFeature.sourceIdentifier, type: GeoJSONSource.self) {
+                try style?.updateGeoJSONSource(withId: trackStyledFeature.sourceIdentifier,
+                                               geoJSON: Feature(geometry: .lineString(trackStyledFeature.lineString)))
+            }
+            
+            if let _ = try style?.source(withId: rawTrackStyledFeature.sourceIdentifier, type: GeoJSONSource.self) {
+                try style?.updateGeoJSONSource(withId: rawTrackStyledFeature.sourceIdentifier,
+                                               geoJSON: Feature(geometry: .lineString(rawTrackStyledFeature.lineString)))
+            }
+        } catch {
+            NSLog("Error occured while performing operation with source: \(error.localizedDescription).")
+        }
     }
     
     func addStyledFeature(_ styledFeature: StyledFeature) {
-        var source = GeoJSONSource()
-        source.data = .geometry(.lineString(styledFeature.lineString))
-        _ = navigationMapView.mapView.style.addSource(source: source,
-                                                      identifier: styledFeature.sourceIdentifier)
-        
-        var layer = LineLayer(id: styledFeature.layerIdentifier)
-        layer.source = styledFeature.sourceIdentifier
-        layer.paint?.lineWidth = .constant(styledFeature.lineWidth)
-        layer.paint?.lineColor = .constant(.init(color: styledFeature.color))
-        _ = navigationMapView.mapView.style.addLayer(layer: layer)
+        do {
+            var source = GeoJSONSource()
+            source.data = .geometry(.lineString(styledFeature.lineString))
+            try navigationMapView.mapView.style.addSource(source,
+                                                          id: styledFeature.sourceIdentifier)
+            
+            var layer = LineLayer(id: styledFeature.layerIdentifier)
+            layer.source = styledFeature.sourceIdentifier
+            layer.paint?.lineWidth = .constant(styledFeature.lineWidth)
+            layer.paint?.lineColor = .constant(.init(color: styledFeature.color))
+            try navigationMapView.mapView.style.addLayer(layer)
+        } catch {
+            NSLog("Failed to perform operation with error: \(error.localizedDescription).")
```

Comment:
There are log statements in response to runtime styling errors throughout this PR. Is this is a best practice, or should we do something more aggressive like asserting or rethrowing? How likely are these errors?

Reply:
In the past I've been frustrated by how fatal the Maps SDK treats things like adding a duplicate style identifier, etc. so I would have liked to see this kind of non-fatal error logging.

That said, it might be useful to make these errors more fatal at least for this period when the Maps SDK v10 is still in beta. it might help us to catch issues in the SDK that might otherwise slip by unnoticed.

Reply:
I was thinking about keeping those log statements to see if there are any issues during development process in runtime styling, when our styling logic is more stable we can probably remove it. Also as @avi-c found we currently have some errors already, so it'd be good to create a ticket to tackle those.